### PR TITLE
QC-QMC chemistry

### PR DESCRIPTION
**Pull request by yitchen-tim**
*Description of changes:*
Example and code for Quantum_Computing_Quantum_Monte_Carlo (QC-QMC)

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws-samples/amazon-braket-algorithm-library/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws-samples/amazon-braket-algorithm-library/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws-samples/amazon-braket-algorithm-library/blob/main/README.md) and [API docs](https://github.com/aws-samples/amazon-braket-algorithm-library/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by yitchen-tim** [comment]
The QMC algorithm is ready for review. pyscf package that caused issue in MacOS checks is replaced by pennylane.qchem

**Review by mbeach-aws — DISMISSED** [review]

File: src/braket/experimental/algorithms/qc_qmc/classical_qmc.py [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+
+@dataclass
+class ChemicalProperties:
+    h1e: np.ndarray  # one-body term
+    eri: np.ndarray  # two-body term
+    nuclear_repulsion: float  # nuclear repulsion energy
+    v_0: np.ndarray  # one-body term stored as np.ndarray with mean-field subtraction
+    h_chem: np.ndarray  # one-body term stored as np.ndarray, without mean-field subtraction
+    v_gamma: List[np.ndarray]  # 1j * L_gamma
+    L_gamma: List[np.ndarray]  # Cholesky vector decomposed from two-body terms
```
Comment (ajberdy) [inline src/braket/experimental/algorithms/qc_qmc/classical_qmc.py:21]: I would love to see a style guide or some kind of source encouraging breaking code style for established scientific names. For now, I'm not convinced that it's worth the deviation

File: src/braket/experimental/algorithms/qc_qmc/qc_qmc.py [hunk 2]
```diff
@@ -0,0 +1,536 @@
...
+    ChemicalProperties,
+    greens_pq,
+    hartree_fock_energy,
+    imag_time_propogator,
+    local_energy,
+    propagate_walker,
+    reortho,
+)
+
+np.seterr(divide="ignore", invalid="ignore")  # ignore divide by zero
```
Comment (ajberdy) [inline src/braket/experimental/algorithms/qc_qmc/qc_qmc.py:19]: curious why we're doing this

File: test/unit_tests/braket/experimental/algorithms/qc_qmc/test_qc_qmc.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.41729459]], requires_grad=False)
+    mol = qml.qchem.Molecule(symbols, geometry, basis_name="sto-3g")
+    trial = np.array([[1, 0], [0, 1], [0, 0], [0, 0]])
+    prop = chemistry_preparation(mol, geometry, trial)
+    dev = qml.device("lightning.qubit", wires=4)
+    e_hf = hartree_fock_energy(trial, prop)
+    return (trial, prop, dev, e_hf)
+
+
+def trial_state_circuit() -> None:
```
Comment (ajberdy) [inline test/unit_tests/braket/experimental/algorithms/qc_qmc/test_qc_qmc.py:30]: nit: inconsistent about -> None in this file. Not convinced it's necessary for tests

Comment (ajberdy) [reply test/unit_tests/braket/experimental/algorithms/qc_qmc/test_qc_qmc.py:30]: Actually nvm, I see this isn't a test function. Seems good to me then

File: notebooks/advanced_algorithms/Quantum_Computing_Quantum_Monte_Carlo.ipynb [hunk 4]
```diff
@@ -0,0 +1,261 @@
+{
```
Comment (ajberdy) [inline notebooks/advanced_algorithms/Quantum_Computing_Quantum_Monte_Carlo.ipynb:1]: <p>Nit: local simulator usually refers to the Braket local simulator. Might be worth specifying local lightning simulator</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws-samples/amazon-braket-algorithm-library/pull/19/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Review by ajberdy — DISMISSED** [review]
Approved with comments.

**Review by ajberdy — APPROVED** [review]

**Pull request merged**